adapters/handlers/grpc/v1/batch/scheduler.go

```diff
@@ -0,0 +1,154 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2025 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package batch
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	enterrors "github.com/weaviate/weaviate/entities/errors"
+	pb "github.com/weaviate/weaviate/grpc/generated/protocol/v1"
+)
+
+type Scheduler struct {
+	logger        logrus.FieldLogger
+	writeQueues   *WriteQueues
+	internalQueue internalQueue
+}
+
+func NewScheduler(writeQueues *WriteQueues, internalQueue internalQueue, logger logrus.FieldLogger) *Scheduler {
+	return &Scheduler{
+		logger:        logger,
+		writeQueues:   writeQueues,
+		internalQueue: internalQueue,
+	}
+}
+
+func (s *Scheduler) Loop(ctx context.Context) {
+	ticker := time.NewTicker(100 * time.Millisecond)
+	for {
+		select {
+		case <-ctx.Done():
```

Comment:
we need to remember to stop a ticker here in this case also to prevent resource leaks